sdk/communication/AzureCommunication/Tests/CommunicationPolicyTokenCredentialTests.swift

```diff
@@ -0,0 +1,148 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import XCTest
+
+#if canImport(AzureCommunication)
+@testable import AzureCommunication
+#endif
+#if canImport(AzureCore)
+@testable import AzureCore
+#endif
+
+class CommunicationPolicyTokenCredentialTests: XCTestCase {
+    let sampleToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjMyNTAzNjgwMDAwfQ.9i7FNNHHJT8cOzo-yrAUJyBSfJ-tPPk2emcHavOEpWc"
+    let aampleTokenExpiry = 32503680000
+    let expiredToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjEwMH0.1h_scYkNp-G98-O4cW6KvfJZwiz54uJMyeDACE4nypg"
+            
+    func fetchTokenSync(completionHandler: TokenRefreshOnCompletion) {
+        let newToken = sampleToken
+        completionHandler(newToken, nil)
+    }
+    
+    func testCreateStaticToken() throws {
+        let expectation = self.expectation(description: "Create static token")
+        
+        let token = expiredToken
+        let userCredential = try CommunicationTokenCredential(token: token)
+        let communicationTokenPolicy = CommunicationPolicyTokenCredential(userCredential)
+        communicationTokenPolicy.token(forScopes: [""]) { (accessToken, error) in
+            XCTAssertNil(error)
+            XCTAssertNotNil(accessToken)
+            expectation.fulfill()
+        }
+        
+        waitForExpectations(timeout: 1000) { (error) in
+            if let error = error {
+                XCTFail("Create token timed out: \(error)")
+            }
+        }
+    }
+    
+    func testCreateRefreshableWithoutInitialToken() throws {
+        let expectation = self.expectation(description: "Create refreshable without initial token")
+        let options = CommunicationTokenRefreshOptions(
+            refreshProactively: true,
+            tokenRefresher: fetchTokenSync
+            )
+        let userCredential = try CommunicationTokenCredential(with: options)
+        let communicationTokenPolicy = CommunicationPolicyTokenCredential(userCredential)
+        
+        communicationTokenPolicy.token(forScopes: [""]) { (accessToken, error) in
+            XCTAssertNil(error)
+            XCTAssertNotNil(accessToken)
+            expectation.fulfill()
+        }
+        
+        waitForExpectations(timeout: 1000) { (error) in
+            if let error = error {
+                XCTFail("Create token timed out: \(error)")
+            }
+        }
+    }
+    
+    func testCreateRefreshableWithInitialToken() throws {
+        let expectation = self.expectation(description: "Create refreshable with initial token")
+        let token = expiredToken
+        let options = CommunicationTokenRefreshOptions(
```

Comment:
@JoshuaLai If you can review these tests still look ok, I just had to make a couple updates to use the CommunicationTokenRefreshOptions

Reply:
Yes everything still looks good